### PR TITLE
Hyperband/bayesopt cleanup

### PR DESCRIPTION
**Pull request by gradientsky**
*Description of changes:*
Code cleanup after bayesopt/hpo removal

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-1488-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1488/1/index.html

File: core/src/autogluon/core/searcher/searcher_factory.py [hunk 1]
```diff
@@ -18,44 +18,25 @@ def searcher_factory(searcher_name, **kwargs):
...
-    scheduler (see FIFOScheduler), which provides most of the required kwargs.
-
-    Note: RLSearcher is not supported here, because its **kwargs are different
-    from other searchers (no 'configspace').
-
-    The 'bayesopt' searchers depend on which scheduler is used (GPFIFOSearcher
-    for FIFOScheduler, GPMultiFidelitySearcher for HyperbandScheduler). They
-    have many additional parameters in kwargs (see docstrings for
-    GPFIFOSearcher, GPMultiFidelitySearcher).
+    scheduler (see LocalRandomSearcher), which provides most of the required kwargs.
```
Comment (Innixma) [inline core/src/autogluon/core/searcher/searcher_factory.py:21]: LocalSequentialScheduler

File: vision/src/autogluon/vision/detector/detector.py [hunk 2]
```diff
@@ -376,8 +375,8 @@ def _validate_kwargs(self, kwargs):
         hpo_tune_args = kwargs.get('hyperparameter_tune_kwargs', {})
         hpo_tune_args['num_trials'] = hpo_tune_args.get('num_trials', 1)
         hpo_tune_args['searcher'] = hpo_tune_args.get('searcher', 'random')
-        if not hpo_tune_args['searcher'] in ('random', 'bayesopt', 'grid'):
-            raise ValueError(f"Invalid searcher: {hpo_tune_args['searcher']}, supported: ('random', 'bayesopt', 'grid')")
+        if not hpo_tune_args['searcher'] in ('random', 'grid'):
+            raise ValueError(f"Invalid searcher: {hpo_tune_args['searcher']}, supported: ('random', 'grid')")
         hpo_tune_args['scheduler'] = hpo_tune_args.get('scheduler', 'local')
         if not hpo_tune_args['scheduler'] in ('fifo', 'local'):
             raise ValueError(f"Invalid searcher: {hpo_tune_args['searcher']}, supported: ('fifo', 'local')")
```
Comment (Innixma) [inline vision/src/autogluon/vision/detector/detector.py:382]: supported: ('local')

**Review by Innixma — APPROVED** [review]
LGTM! Added a couple minor comments

**Pull request merged**